app/shop.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+from typing import Any
+
+from app.point import Point
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: Point,
+            products: dict[str, float | int]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @classmethod
+    def create_shop(
+            cls,
+            shop_info: dict[str, str | list[int] | dict[str, (int | float)]]
+    ) -> Shop:
+        return cls(
+            name=shop_info["name"],
+            location=Point.create_from_location(shop_info["location"]),
+            products=shop_info["products"]
```

Comment:
Ensure that the keys 'name', 'location', and 'products' are always present in the `shop_info` dictionary to avoid KeyError. Consider using the `get` method with default values or handling exceptions.